### DISCUSSION TITLE
RAO postprocess library

### DISCUSSION THREAD

**Discussion by baptistelabat-syroco** [capytaine/capytaine#599] [Q&A]
Hi, 

I am looking for some existing open source libraries to postprocess the results of Capytaine. I was using closed source software for this in a previous company. Here is the list of features I am looking for (might not be difficult to redo, but better to have a library):

- Use of force or motion RAO
- Plotting RAO and navigating
- Estimation of viscous roll damping
- Ability to transfer motion RAO from one reference point to another one
- Ability to compute speed and acceleration RAO from motion RAO
- Ability to compute free surface elevation RAO and relative motion RAO
- Merging RAO with coarse and fine resolution into one (interpolation)
- Interface with some 2D wave spectra
- Ability to compute response spectrum in irregular sea
- Computation of moments of spectra
- Computation of extrema

I have found https://github.com/arnavdoss/pyRAO, but it is not well packaged to be used as a library.
I have found https://github.com/4Subsea/waveresponse-python, but I would like to be able to work directly with xarray capytaine output
I have found https://github.com/RubendeBruin/mafredo, but it seems rather limited in its 

Any help welcome

**Comment by baptistelabat-syroco** [comment]
For your information, I was able to do most of the things I wanted with waveresponse-python. 
And for preprocessing of the mesh, I was able to do nice things with pymeshlab (using isotropic explicit remeshing)

**Reply by mancellin** [reply]
Thank you for sharing this.
I would never have noticed PyMeshLab remeshing algorithm in their documentation otherwise.

Here is a code sample of usage with Capytaine if anybody is interested:
```python
import numpy as np
import capytaine as cpt
import pymeshlab

def remeshing(mesh, edge_size):
    ms = pymeshlab.MeshSet()
    ms.add_mesh(pymeshlab.Mesh(mesh.vertices, mesh.faces))
    ms.meshing_isotropic_explicit_remeshing(targetlen=pymeshlab.PureValue(edge_size))
    m = ms.current_mesh()
    f = m.face_matrix()
    f = np.concat([f, f[:, [2]]], axis=1)
    return cpt.Mesh(m.vertex_matrix(), f)

mesh = cpt.mesh_sphere(radius=1.0, resolution=(40, 40)).immersed_part()
mesh = remeshing(mesh, edge_size=0.1)
mesh.show()

```
